test/e2e/storage/testsuites/provisioning.go

```diff
@@ -73,132 +80,81 @@ func (p *provisioningTestSuite) getTestSuiteInfo() TestSuiteInfo {
 	return p.tsInfo
 }
 
-func (p *provisioningTestSuite) skipUnsupportedTest(pattern testpatterns.TestPattern, driver TestDriver) {
+func (p *provisioningTestSuite) isTestSupported(pattern testpatterns.TestPattern, driver TestDriver) bool {
+	_, ok := driver.(DynamicPVTestDriver)
+	return ok
 }
 
-func createProvisioningTestInput(driver TestDriver, pattern testpatterns.TestPattern) (provisioningTestResource, provisioningTestInput) {
-	// Setup test resource for driver and testpattern
-	resource := provisioningTestResource{}
-	resource.setupResource(driver, pattern)
-
-	input := provisioningTestInput{
-		testCase: StorageClassTest{
-			ClaimSize:    resource.claimSize,
-			ExpectedSize: resource.claimSize,
-		},
-		cs:    driver.GetDriverInfo().Config.Framework.ClientSet,
-		pvc:   resource.pvc,
-		sc:    resource.sc,
-		dInfo: driver.GetDriverInfo(),
-	}
-
-	if driver.GetDriverInfo().Config.ClientNodeName != "" {
-		input.testCase.NodeName = driver.GetDriverInfo().Config.ClientNodeName
-	}
-
-	return resource, input
-}
-
-func (p *provisioningTestSuite) execTest(driver TestDriver, pattern testpatterns.TestPattern) {
+func (p *provisioningTestSuite) setupTest(driver TestDriver, config *TestConfig, pattern testpatterns.TestPattern) {
 	Context(getTestNameStr(p, pattern), func() {
 		var (
-			resource     provisioningTestResource
-			input        provisioningTestInput
-			needsCleanup bool
+			testCase StorageClassTest
+			pvc      *v1.PersistentVolumeClaim
+			sc       *storage.StorageClass
 		)
 
-		BeforeEach(func() {
-			needsCleanup = false
-			// Skip unsupported tests to avoid unnecessary resource initialization
-			skipUnsupportedTest(p, driver, pattern)
-			needsCleanup = true
-
-			// Create test input
-			resource, input = createProvisioningTestInput(driver, pattern)
-		})
+		Expect(pattern.VolType).To(Equal(testpatterns.DynamicPV))
 
-		AfterEach(func() {
-			if needsCleanup {
-				resource.cleanupResource(driver, pattern)
+		BeforeEach(func() {
+			dDriver := driver.(DynamicPVTestDriver) // Checked in isTestSupported.
+			claimSize := dDriver.GetClaimSize()
+			sc = dDriver.GetDynamicProvisionStorageClass(config, "")
+			if sc == nil {
+				framework.Skipf("Driver %q does not define Dynamic Provision StorageClass - skipping", driver.GetDriverInfo().Name)
+			}
+			pvc = getClaim(claimSize, config.Framework.Namespace.Name)
```

Comment:
nit: still think this stuff should go in a function. Maybe not `setupResource` exactly but `setupStorageClassTest` or something?